src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -121,8 +121,16 @@ public RMProfileClient(AzureRMProfile profile)
                         newTenant == null &&                         
                         TryGetTenantSubscription(token, account, environment, tenant, subscriptionId, subscriptionName, out tempSubscription, out tempTenant))
                     {
-                        newTenant = tempTenant;
-                        newSubscription = tempSubscription;
+                        if (tempSubscription == null && i + 1 < tenants.Count())
```

Comment:
@hovsepm 
- yes.
- To keep the tenant selecting order, how about iterating tenants in a reverse fashion? Personally, I don't think "first tenant/subscription" makes much sense here if tenants are not sorted(or sorted alphabetically).